Taskfile.yml

```diff
@@ -0,0 +1,193 @@
+    cmds:
+    - ./bin/gofumpt
+    - go generate -x ./...
+    - task: fmt
+
+  fmt:
+    desc: "Format code"
+    cmds:
+#      - ./bin/gofumpt{{.EXT}}
+      - ./bin/gofumpt{{.EXT}} -w .
+
+  test:
+    cmds:
+      - go test -race -shuffle=on -coverprofile=cover.txt -coverpkg=./... ./...
+      - go test -race -shuffle=on -bench=. -benchtime=1x ./...
+
+
+  bench-short:
+    desc: "Benchmark for about 20 seconds (with default BENCHTIME)"
+    cmds:
+      - go test -list='Benchmark.*' ./...
+#      - {{.RM}} new.txt
+      # Running four functions for $(BENCHTIME) each..."
+      - go test -bench=BenchmarkArray    -benchtime={{.BENCHTIME}} ./internal/bson/  | tee -a new.txt
+      - go test -bench=BenchmarkDocument -benchtime={{.BENCHTIME}} ./internal/bson/  | tee -a new.txt
+      - go test -bench=BenchmarkArray    -benchtime={{.BENCHTIME}} ./internal/fjson/ | tee -a new.txt
+      - go test -bench=BenchmarkDocument -benchtime={{.BENCHTIME}} ./internal/fjson/ | tee -a new.txt
+      - bin/benchstat old.txt new.txt
+
+
+
+  # That's not quite correct: https://github.com/golang/go/issues/15513
+  # But good enough for us.
+  fuzz-init:
+    deps:
+      - gen-version
+    cmds:
+      - go test -count=0 ./...
+
+  # Those commands should still run tests (i.e., should not have -run=XXX flags)
+  # to fill seed corpus for fuzz tests that use WriteSeedCorpusFile (e.g., FuzzHandler).
+  fuzz:
+    desc: "Fuzz for about 2 minutes (with default FUZZTIME)"
+    cmds:
+      - go test -list='Fuzz.*' ./...
+      # Running eight functions for {{.FUZZTIME}} each..."
+      - go test -fuzz=FuzzArray -fuzztime={{.FUZZTIME}} ./internal/bson/
+      - go test -fuzz=FuzzDocument -fuzztime={{.FUZZTIME}} ./internal/bson/
+      - go test -fuzz=FuzzArray -fuzztime={{.FUZZTIME}} ./internal/fjson/
+      - go test -fuzz=FuzzDocument -fuzztime={{.FUZZTIME}} ./internal/fjson/
+      - go test -fuzz=FuzzMsg -fuzztime={{.FUZZTIME}} ./internal/wire/
+      - go test -fuzz=FuzzQuery -fuzztime={{.FUZZTIME}} ./internal/wire/
+      - go test -fuzz=FuzzReply -fuzztime={{.FUZZTIME}} ./internal/wire/
+      - go test -fuzz=FuzzHandler -fuzztime={{.FUZZTIME}} ./internal/handlers/
+
+  fuzz-corpus:
+    desc: "Sync seed and generated fuzz corpora with FUZZCORPUS"
+    cmds:
+      - go run ./cmd/fuzztool/fuzztool.go -dst={{.FUZZCORPUS}} -src=generated
+      - go run ./cmd/fuzztool/fuzztool.go -dst={{.FUZZCORPUS}} -src=seed
+      - go run ./cmd/fuzztool/fuzztool.go -src={{.FUZZCORPUS}} -dst=generated
+
+  build-testcover:
+    desc: "Build bin/ferretdb-testcover"
+    cmds:
+      - go test -c -o=bin/ferretdb-testcover -trimpath -tags=testcover -race -coverpkg=./... ./cmd/ferretdb
+
+  run:
+    desc: "Run FerretDB"
+    deps:
+      - build-testcover
+    cmds:
+      - bin/ferretdb-testcover -test.coverprofile=cover.txt -mode=diff-normal -listen-addr=:27017
+
+  lint:
+    desc: "Run linters"
+    cmds:
+      - bin/go-sumtype bin/golangci-lint ## Run linters
```

Comment:
That was a dependency, not a command: `make lint` depends on `bin/go-sumtype` and `bin/golangci-lint` targets.

Reply:
So it was to check wheter this tool installed, right?

Reply:
```
lint: bin/go-sumtype
  cmd1 ...

bin/go-sumtype:
  cmd2 ...
```

When invoked as `make lint`, GNU Make first checks if `bin/go-sumtype` file exist (and "fresh"). If it doesn't, it invokes  `bin/go-sumtype` Makefile target first, running `cmd2 ...`.

The current Taskfile invokes `bin/go-sumtype` with argument `bin/golangci-lint` which does nothing useful

(please re-request review if review's attention is needed)

Reply:
After digging into docs and some researching I've ended up with adding preconditions that would check files existance and output error which would look like this:
```sh
task: Please run `task init` first
task: precondition not met
```
Every other options would make tasks more complicated.